src/VisualStudio/CSharp/Impl/ProjectSystemShim/TempPECompilerService.cs

```diff
@@ -70,17 +72,31 @@ private CSharpCommandLineArguments ParseCommandLineArguments(string baseDirector
 
             for (int i = 0; i < optionNames.Length; i++)
             {
-                if (optionNames[i] == "r")
+                var optionName = optionNames[i];
+                var optionValue = optionValues[i];
+
+                if (optionName == "r")
                 {
                     // We get a pipe-delimited list of references, so split them back apart
-                    foreach (var reference in ((string)optionValues[i]).Split(new[] { '|' }, StringSplitOptions.RemoveEmptyEntries))
+                    foreach (var reference in ((string)optionValue).Split(new[] { '|' }, StringSplitOptions.RemoveEmptyEntries))
```

Comment:
(Sorry as I don't know the code)

Is this ever user input? If so, can I pass /r:[nothing] and split will return an array with one element.